### PR TITLE
TZUP-274:-This is the backend code to support requisition only flag on requisition template

### PR DESCRIPTION
**Pull request by hassanhussein**
This the backend code for the TZUP-274: requisition report only functionality

[TZUP-274]: https://openlmis.atlassian.net/browse/TZUP-274?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/openlmis/openlmis-requisition/62)

### CODE REVIEW

File: src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -391,4 +398,41 @@ private AvailableRequisitionColumn getColumnWithOption() {
   private AvailableRequisitionColumnOption getOption(String uuid) {
     return availableRequisitionColumnOptionRepository.findById(UUID.fromString(uuid)).orElse(null);
   }
+
+  @Test
+  public void shouldAllowDuplicationForRequisitionReportOnlyTemplateNames() {
+
```
Comment (anawrotsoldevelo) [inline src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java:404]: Empty lines at the beginning and at the end of the functions are inconsistent with the rest of the code. Please delete them in all added functions

File: src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -391,4 +398,41 @@ private AvailableRequisitionColumn getColumnWithOption() {
...
+    RequisitionTemplate template = generateInstance();
+    template.requisitionReportingOnly();
+    repository.saveAndFlush(template);
+
+  }
+
```
Comment (anawrotsoldevelo) [inline src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java:410]: Other functions are separated only with single line, please delete this extra line

File: src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java (hunk 1 above)
Comment (hassanhussein) [reply src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java]: This has been resolved, please see the updated one.

File: src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java (hunk 2 above)
Comment (hassanhussein) [reply src/integration-test/java/org/openlmis/requisition/repository/RequisitionTemplateRepositoryIntegrationTest.java:408]: Resolved

**Review by anawrotsoldevelo — APPROVED** [review]

**Pull request merged**